gnome-image-installer/util/gis-dmi.c

```diff
@@ -0,0 +1,112 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
+/*
+ * Copyright © 2018 Endless Mobile, Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
+ * 02111-1307, USA.
+ *
+ */
+
+#include "gis-dmi.h"
+#include "config.h"
+
+#include <gio/gio.h>
+
+#define SYS_VENDOR "/sys/class/dmi/id/sys_vendor"
+#define SYS_PRODUCT "/sys/class/dmi/id/product_name"
+
+/**
+ * gis_dmi_sanitize_string:
+ * @string: (transfer none): an arbitrary bytestring
+ *
+ * Returns: (transfer full): a copy of @string, with bytes outside the ASCII
+ *  printable range removed, or %NULL if @string is %NULL or contains no
+ *  printable characters
+ */
+gchar *
+gis_dmi_sanitize_string (const gchar *string)
+{
+  g_autofree gchar *sanitized = g_strdup (string);
+  const gchar *r = string;
+  gchar *w = sanitized;
+
+  if (string == NULL)
+    return NULL;
+
+  for (; *r != '\0'; r++)
+    {
+      if (*r < 32 || *r > 126)
```

Comment:
To me `126` seems more significant than `'~'` because I don't know the ASCII codepoint for tilde off-hand, but I do know that 127 is the highest codepoint defined in ASCII and is DEL.

Reply:
🤷‍♂️